### PR TITLE
PLAT-122792: Fixed LabeledIconButton: Huge Icon is cut off

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A CHANGELOG entry is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Issue: agate/LabeledIconButton: Huge size icon with label position `after` or `right` is cut off. 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Solution: Removed marquee from LabeledIconButton, adjusted component max-with.

### Links
[//]: # (Related issues, references)
PLAT-122792

### Comments

### CODE REVIEW

File: LabeledIconButton/LabeledIconButton.js [hunk 1]
```diff
@@ -12,12 +12,15 @@
 
 import EnactPropTypes from '@enact/core/internal/prop-types';
 import kind from '@enact/core/kind';
+import Spottable from '@enact/spotlight/Spottable';
+import {ButtonDecorator as UiButtonDecorator} from '@enact/ui/Button';
 import {LabeledIconBase as UiLabeledIconBase, LabeledIconDecorator as UiLabeledIconDecorator} from '@enact/ui/LabeledIcon';
+import Pure from '@enact/ui/internal/Pure';
```
Comment (daniel-stoian-lgp) [inline LabeledIconButton/LabeledIconButton.js:18]: sort imports

**Review by daniel-stoian-lgp — APPROVED** [review]

**Comment by ybsung** [comment]
There is one regression.
If passing null string (''), then the icon sticks to the left.

AS-IS
image

TO-BE
image

File: LabeledIconButton/LabeledIconButton.js [hunk 2]
```diff
@@ -163,7 +166,10 @@ const LabeledIconButtonBase = kind({
  */
 const LabeledIconButtonDecorator = compose(
 	UiLabeledIconDecorator,
-	ButtonDecorator
+	Pure,
+	UiButtonDecorator,
+	Spottable,
+	Skinnable
```
Comment (ybsung) [inline LabeledIconButton/LabeledIconButton.js:172]: `ButtonDecorator` is replaced with 
```
	Pure,
	UiButtonDecorator,
	Spottable,
	Skinnable
```

By doing it, I'm worried that it is ok to remove `IconButtonDecorator`, `TooltipDecorator({tooltipDestinationProp: 'decoration'})`.

CC: @jeonghee27

Comment (jeonghee27) [reply LabeledIconButton/LabeledIconButton.js:172]: If LabeledIconButton don't need the iconOnly prop, IconButtonDecorator is not needed.
But if the tooltip support has not changed, let's keep TooltipDecorator.

**Comment by stanca-pop-lgp** [comment]
@ybsung Icon sticks to the left when there are no children because LabeledIconButton has max-width: min-content;. Is there a reason why a LabeledIconButton should take more space than its icon if there are no children?
More than that, is there a reason why a LabeledIconButton would have no label? 
Thank you.

**Comment by ybsung** [comment]
Hello @stanca-pop-lgp 

> @ybsung Icon sticks to the left when there are no children because LabeledIconButton has max-width: min-content;. Is there a reason why a LabeledIconButton should take more space than its icon if there are no children?
More than that, is there a reason why a LabeledIconButton would have no label?
Thank you.

The reason why I mentioned before is

First of all, if there is no special reason, it might be better to preserve component behavior and look. There were some space to the left before. But I found that there is no space in develop branch, too. So it would be better to fix it in a separate issue.

Second, as you mentioned, the LabeledIconButton  is usually used with a label, if without a label, the IconButton might be used. But app developers could pass the `children` prop asynchronously. So in my opinion, it might be better to keep the Button layout even though some props has been arrived yet or aren't passed.

In addition, it would be helpful to check any JS doc parse errors. I pushed one commit to fix some errors including one which is not related with your PR.

**Review by jeonghee27 — COMMENTED** [review]
LGTM

**Pull request merged**